From 9b93bb6b3c4d2d13a93693b6201372710b8d7d02 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Mon, 3 Dec 2018 08:58:36 +0800
Subject: [PATCH] drm/rockchip: tve: Add tve shutdown interface

When system shutdown, shutdown interface will be called.
TVE should be disabled when system shutdown.

Change-Id: If9dd29605cc0bd67aa8e9c494a98f89e625e4c50
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_tve.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_tve.c b/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
index aab9c2e95fb3..50b8e3631753 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
@@ -579,6 +579,18 @@ static int rockchip_tve_probe(struct platform_device *pdev)
 	return 0;
 }
 
+static void rockchip_tve_shutdown(struct platform_device *pdev)
+{
+	struct rockchip_tve *tve = dev_get_drvdata(&pdev->dev);
+
+	mutex_lock(&tve->suspend_lock);
+
+	dev_dbg(tve->dev, "tve shutdown\n");
+	cvbs_set_disable(tve);
+
+	mutex_unlock(&tve->suspend_lock);
+}
+
 static int rockchip_tve_remove(struct platform_device *pdev)
 {
 	component_del(&pdev->dev, &rockchip_tve_component_ops);
@@ -589,6 +601,7 @@ static int rockchip_tve_remove(struct platform_device *pdev)
 struct platform_driver rockchip_tve_driver = {
 	.probe = rockchip_tve_probe,
 	.remove = rockchip_tve_remove,
+	.shutdown = rockchip_tve_shutdown,
 	.driver = {
 		   .name = "rockchip-tve",
 		   .of_match_table = of_match_ptr(rockchip_tve_dt_ids),
-- 
2.35.3

